apis/apps/v1alpha1/cluster_types.go

```diff
@@ -868,6 +868,15 @@ type ClusterComponentSpec struct {
 	// +optional
 	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.10.0"
 	Monitor *bool `json:"monitor,omitempty"`
+
+	// Specifies the user-defined configuration template or parameters.
+	//
+	// +patchMergeKey=name
+	// +patchStrategy=merge,retainKeys
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	ComponentConfigItems []ComponentConfigItem `json:"componentConfigItem,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"name"`
```

Comment:
Should be named to "parameters". 

There are two kinds of parameters:
- Parameters reside in config files. And the relationship of paramters and corresponding config file should already be described in ConfigConstraint (will be renamed to ParameterDescriptors or ParameterMetadata).
- Parameters that 're not persisted in config files. e.g. They may be stored in a system table in the database.

So in the Cluster API, we should not expose the relationship of parameters and config file. And the data structure should look this:

struct ComponentParameters {
    // name of ConfigConstraint, where the parameters 're defined
    Name string `json:"name"`

    Parameters map[string]*string `json:"pairs,omitempty"`
}

Reply:
update api